src/python_testing/TC_RVCCLEANM_2_2.py

```diff
@@ -0,0 +1,146 @@
+        self.run_mode_dut = 0
+        self.old_clean_mode_dut = 0
+        self.new_clean_mode_th = 0
+
+    async def read_mod_attribute_expect_success(self, cluster, attribute):
+        return await self.read_single_attribute_check_success(
+            endpoint=self.endpoint, cluster=cluster, attribute=attribute)
+
+    async def read_run_supported_modes(self) -> Clusters.Objects.RvcRunMode.Attributes.SupportedModes:
+        ret = await self.read_mod_attribute_expect_success(
+            Clusters.RvcRunMode,
+            Clusters.RvcRunMode.Attributes.SupportedModes)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcRunMode.Attributes.SupportedModes), "")
+        return ret
+
+    async def read_clean_supported_modes(self) -> Clusters.Objects.RvcCleanMode.Attributes.SupportedModes:
+        ret = await self.read_mod_attribute_expect_success(
+            Clusters.RvcCleanMode,
+            Clusters.RvcCleanMode.Attributes.SupportedModes)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcCleanMode.Attributes.SupportedModes), "")
+        return ret
+
+    async def send_change_to_mode_cmd(self, newMode) -> Clusters.Objects.RvcCleanMode.Commands.ChangeToModeResponse:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.RvcCleanMode.Commands.ChangeToMode(newMode=newMode), endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.RvcCleanMode.Commands.ChangeToModeResponse),
+                            "Unexpected return type for ChangeToMode")
+        return ret
+
+    # Prints the instruction and waits for a user input to continue
+    def print_instruction(self, step_number, instruction):
+        self.print_step(step_number, instruction)
+        input("Press Enter when done.\n")
+
+    @async_test_body
+    async def test_TC_RVCCLEANM_2_2(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+
+        self.endpoint = self.matter_test_config.global_test_params['PIXIT_ENDPOINT']
+
+        asserts.assert_true(self.check_pics("RVCCLEANM.S"), "RVCCLEANM.S must be supported")
+        asserts.assert_true(self.check_pics("RVCRUNM.S.A0000"), "RVCRUNM.S.A0000 must be supported")
+        asserts.assert_true(self.check_pics("RVCRUNM.S.A0001"), "RVCRUNM.S.A0001 must be supported")
+
+        self.print_step(1, "Commissioning, already done")
+
+        self.print_instruction(2, "Manually put the device in a state in which the RVC Run Mode "
+                                  "cluster’s CurrentMode attribute is set to a mode without the Idle mode tag.")
+
+        self.print_step(3, "Read the RvcRunMode SupportedModes attribute")
+        supported_run_modes = await self.read_run_supported_modes()
+        for mode in supported_run_modes:
```

Comment:
I totally don't get what's going on here...what's mode.mode? why is supported_run_modes and self.supported_run_modes_dut different?

Reply:
The `supported_run_modes` is a dictionary of ModeOptionStructs with the mode value as the key. A copy of this is kept to enable some of the later steps. The `supported_run_mode_dut` is a basically the list of keys in that dictionary. It was named as such to match closely to the test plan wording.